src/write.c

```diff
@@ -1440,74 +1613,79 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
     }                                                                      //
-    avifRWStreamWriteChars(&s, "mif1", 4);                                 // ... compatible_brands[]
-    avifRWStreamWriteChars(&s, "miaf", 4);                                 // ... compatible_brands[]
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "mif1", 4));                  // ... compatible_brands[]
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "miaf", 4));                  // ... compatible_brands[]
     if ((imageMetadata->depth == 8) || (imageMetadata->depth == 10)) {     //
         if (imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {        //
-            avifRWStreamWriteChars(&s, "MA1B", 4);                         // ... compatible_brands[]
+            AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1B", 4));          // ... compatible_brands[]
         } else if (imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) { //
-            avifRWStreamWriteChars(&s, "MA1A", 4);                         // ... compatible_brands[]
+            AVIF_CHECKRES(avifRWStreamWriteChars(&s, "MA1A", 4));          // ... compatible_brands[]
         }
     }
     avifRWStreamFinishBox(&s, ftyp);
 
     // -----------------------------------------------------------------------
     // Start meta
 
-    avifBoxMarker meta = avifRWStreamWriteFullBox(&s, "meta", AVIF_BOX_SIZE_TBD, 0, 0);
+    avifBoxMarker meta;
+    AVIF_CHECKRES(avifRWStreamWriteFullBox(&s, "meta", AVIF_BOX_SIZE_TBD, 0, 0, &meta));
 
     // -----------------------------------------------------------------------
     // Write hdlr
 
-    avifBoxMarker hdlr = avifRWStreamWriteFullBox(&s, "hdlr", AVIF_BOX_SIZE_TBD, 0, 0);
-    avifRWStreamWriteU32(&s, 0);              // unsigned int(32) pre_defined = 0;
-    avifRWStreamWriteChars(&s, "pict", 4);    // unsigned int(32) handler_type;
-    avifRWStreamWriteZeros(&s, 12);           // const unsigned int(32)[3] reserved = 0;
-    avifRWStreamWriteChars(&s, "libavif", 8); // string name; (writing null terminator)
+    avifBoxMarker hdlr;
+    AVIF_CHECKRES(avifRWStreamWriteFullBox(&s, "hdlr", AVIF_BOX_SIZE_TBD, 0, 0, &hdlr));
+    AVIF_CHECKRES(avifRWStreamWriteU32(&s, 0));              // unsigned int(32) pre_defined = 0;
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "pict", 4));    // unsigned int(32) handler_type;
+    AVIF_CHECKRES(avifRWStreamWriteZeros(&s, 12));           // const unsigned int(32)[3] reserved = 0;
+    AVIF_CHECKRES(avifRWStreamWriteChars(&s, "libavif", 8)); // string name; (writing null terminator)
     avifRWStreamFinishBox(&s, hdlr);
 
     // -----------------------------------------------------------------------
     // Write pitm
 
     if (encoder->data->primaryItemID != 0) {
-        avifRWStreamWriteFullBox(&s, "pitm", sizeof(uint16_t), 0, 0);
-        avifRWStreamWriteU16(&s, encoder->data->primaryItemID); //  unsigned int(16) item_ID;
+        avifBoxMarker pitm;
```

Comment:
We can pass `NULL` to `avifRWStreamWriteFullBox()` and avoid the `pitm` variable.